api/store/mongo/tags.go

```diff
@@ -122,11 +102,86 @@ func (s *Store) TagsDelete(ctx context.Context, tenantID string, tag string) (in
 			return int64(0), err
 		}
 
-		return devCount + keyCount + rulCount, nil
+		tagCount, err := s.TagsBulkDeleteTag(sessCtx, tenantID, tag)
+		if err != nil {
+			return int64(0), err
+		}
+
+		return devCount + keyCount + rulCount + tagCount, nil
 	})
 	if err != nil {
 		return int64(0), FromMongoError(err)
 	}
 
 	return count.(int64), nil
 }
+
+func (s *Store) VerifyTagExist(ctx context.Context, tagName, tenant string) (bool, error) {
+	tag := new(models.Tags)
+
+	cursor, err := s.db.Collection("tags").Find(ctx,
+		bson.M{"name": tagName, "tenant_id": tenant})
+	if err != nil {
+		return false, FromMongoError(err)
+	}
+
+	for cursor.Next(ctx) {
+		if err != nil {
+			return false, FromMongoError(err)
+		}
+
+		if tag.Name == "" {
```

Comment:
This check is unnecessary since the presence of a document inherently confirms the tag's existence.